x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -134,6 +136,19 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
+  public async removeKeyPair(): Promise<void> {
```

Comment:
small nit: return `Promise<boolean>` of whether we deleted the key pair and then in `rotateKeyPair` only run `await this.generateKeyPair()` if key pair was removed to save some cycles.